### ISSUE TITLE
Possible false positive in `needless_borrow` for `Mul` implemented for `&T`

### ISSUE THREAD

**Issue by hdevalence** [Manishearth/rust-clippy#1700]
curve25519-dalek has a type `FieldElement`.  There's an implementation of `Mul` for `&FieldElement`, but not `FieldElement`, to avoid a needless copy.

I think this may be a false positive:
```
warning: this expression borrows a reference that is immediately dereferenced by the compiler
   --> src/curve.rs:235:43
    |
235 |         let v_squared: FieldElement = u * &(&(&u.square() + &(&(&constants::A * u) + &one)));
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(needless_borrow)] on by default
    = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#needless_borrow
```

**Comment by oli-obk** [comment]
Hmm.. We ask rustc for the situations where the lint should trigger, so we should not have any false positives. Are you able to remove the ampersand and keep it compiling?

**Comment by hdevalence** [comment]
Oops, indeed, there's just one too many &'s -- sorry for wasting time.
